figures/scripts/fig3-oncoprint-landscape.R

```diff
@@ -0,0 +1,240 @@
+# Oncoprint Landscape Figure
+#
+# 2020
+# Chante Bethell for ALSF - CCDL
+#
+# This script is intended to run steps needed to create Figure 3.
+
+#### Set Up --------------------------------------------------------------------
+
+# Install maftools
+if (!("maftools" %in% installed.packages())) {
+  install.packages("BiocManager")
+  BiocManager::install("maftools")
+}
+library(maftools)
+
+# Load in patchwork for assembling the final multipanel figure
+library(patchwork)
+
+# Get `magrittr` pipe
+`%>%` <- dplyr::`%>%`
+
+#### Directories and Files -----------------------------------------------------
+
+# Detect the ".git" folder -- this will in the project root directory.
+# Use this as the root directory to ensure proper execution, no matter where
+# it is called from.
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# Path to the data obtained via `bash download-data.sh`.
+data_dir <- file.path(root_dir, "data")
+
+# Declare output directory
+output_dir <- file.path(root_dir, "figures", "pngs")
+
+# Source the color palette for plots
+source(
+  file.path(
+    root_dir,
+    "analyses",
+    "oncoprint-landscape",
+    "util",
+    "oncoplot-palette.R"
+  )
+)
+
+#### Read in data --------------------------------------------------------------
+
+# Read in metadata
+metadata <-
+  readr::read_tsv(file.path(root_dir, "data", "pbta-histologies.tsv")) %>%
+  dplyr::rename(Tumor_Sample_Barcode = sample_id)
+
+# Read in MAF files
+maf_df_primary_plus <-
+  readr::read_tsv(
+    file.path(
+      root_dir,
+      "scratch",
+      "oncoprint_files",
+      "all_participants_primary-plus_maf.tsv"
+    )
+  )
+
+maf_df_primary_only <-
+  readr::read_tsv(
+    file.path(
+      root_dir,
+      "scratch",
+      "oncoprint_files",
+      "all_participants_primary-only_maf.tsv"
+    )
+  )
+
+# Read in cnv files
+cnv_file_primary_plus <-
+  readr::read_tsv(
+    file.path(
+      root_dir,
+      "scratch",
+      "oncoprint_files",
+      "all_participants_primary-plus_cnv.tsv"
+    )
+  )
+
+cnv_file_primary_only <-
+  readr::read_tsv(
+    file.path(
+      root_dir,
+      "scratch",
+      "oncoprint_files",
+      "all_participants_primary-only_cnv.tsv"
+    )
+  )
+
+# Read in fusion files
+fusion_file_primary_plus <- 
+  readr::read_tsv(
+    file.path(
+      root_dir,
+      "scratch",
+      "oncoprint_files",
+      "all_participants_primary-plus_fusions.tsv"
+    )
+  )
+
+fusion_file_primary_only <- 
+  readr::read_tsv(
+    file.path(
+      root_dir,
+      "scratch",
+      "oncoprint_files",
+      "all_participants_primary-only_fusions.tsv"
+    )
+  )
+
+# Read in gene list
+gene_list <-
+  readr::read_tsv(
+    file.path(
+      root_dir,
+      "analyses",
+      "interaction-plots",
+      "results",
+      "gene_disease_top50.tsv"
+    )
+  )
+
+#### Custom function ----------------------------------------------------------
+
+prepare_and_plot_oncoprint <- function(maf_df,
```

Comment:
How is this section different or the same from the `01-plot-oncoprint.R`?

Reply:
Can you put a link to  where this code originally comes from? Also is there a way we can minimize the duplication of the same code? (I get that we want to render the final png in one place so the answer to this may be no. )

Reply:
Another idea to be thinking about is if you could make this function into its own `util` file in oncoprint-landscape folder, and then source it. Use it in `oncoprint-landscape` and here. (I did a similar thing for the CDF plotting function if you want to look for an example of that. This makes it so you only have to change this function once in the future and it will be changed in both the final figure and the figures in the `oncoprint-landscape`.

Reply:
> How is this section different or the same from the 01-plot-oncoprint.R?

This section in particular is the same. After making this script I did wonder if there was a way for me to take the two output plots needed from `01-plot-oncoprint.R` and read them into a script to be assembled into a figure panel but have not determined a way to do this as yet.

That being said, your suggestion:
> Another idea to be thinking about is if you could make this function into its own util file in oncoprint-landscape folder, and then source it. Use it in oncoprint-landscape and here. (I did a similar thing for the CDF plotting function if you want to look for an example of that. This makes it so you only have to change this function once in the future and it will be changed in both the final figure and the figures in the oncoprint-landscape.

seems that it will likely be the best solution in this situation.